File: src/renderer/services/skills.ts
import { backend as defaultBackend } from "../backend"
import type { Backend } from "../backend/types"

/** A Claude skill discovered from a `.claude/skills/<name>/SKILL.md` file. */
export interface Skill {
  /** Invocation name (frontmatter `name`, falling back to the directory name). */
  name: string
  /** One-line summary from frontmatter `description`. */
  description: string
  /** Where the skill was found. */
  source: "project" | "user"
}

/**
 * SkillsService discovers the Claude skills available to an agent running in a
 * given workspace. Skills are a Claude-only feature; callers should gate usage
 * on the active agent being "claude".
 */
export class SkillsService {
  private backend: Backend

  constructor(backend: Backend = defaultBackend) {
    this.backend = backend
  }

  /**
   * List skills for a workspace, sorted by name (project shadows user).
   *
   * `claudeConfigDir` is the project's per-account CLAUDE_CONFIG_DIR override, so
   * user skills come from that account's dir. Undefined = the default `~/.claude`.
   */
  async listSkills(workspacePath: string, claudeConfigDir?: string): Promise<Skill[]> {
    return this.backend.invoke<Skill[]>("list_skills", {
      workspacePath,
      claudeConfigDir: claudeConfigDir ?? null,
    })
  }
}

export const skillsService = new SkillsService()

File: src-tauri/src/skills.rs
//! Claude skill discovery for Tauri commands.
//!
//! Thin async wrapper around `overseer_core::skills`.

use overseer_core::paths;
use overseer_core::skills::{self, Skill};

/// List the Claude skills available to an agent running in `workspace_path`.
///
/// Discovers project skills (`<workspace>/.claude/skills`) and user skills
/// (`<config_dir>/skills`); project skills shadow user skills of the same name.
/// `claude_config_dir` is the project's per-account `CLAUDE_CONFIG_DIR` override
/// (raw `~`/`$HOME` allowed); absent = the default `~/.claude`.
#[tauri::command]
pub async fn list_skills(
    workspace_path: String,
    claude_config_dir: Option<String>,
) -> Result<Vec<Skill>, String> {
    let home = paths::get_home_dir().ok();
    let config_dir = paths::resolve_claude_config_dir(claude_config_dir.as_deref(), home.as_deref());
    Ok(skills::list_skills(&workspace_path, config_dir.as_deref()))
}
